### ISSUE TITLE
Recover from defered panic causes crash

### ISSUE THREAD

**Issue by lologarithm** [gopherjs/gopherjs#780]
Issue:
When recovering from a panic created by another deferred function the handler seems to crash.
 
Example here: https://gopherjs.github.io/playground/#/SNlZWoHnci

Execution locally with go run:
```
 ~/c/g/s/s/jsdefer$ go run main.go
recovered in bar: a panic
recovered in foo: bar
Err: failed
```

Execution with gopherjs run:
```
 ~/c/g/s/s/jsdefer$ gopherjs run main.go
recovered in bar: a panic
recovered in foo: bar

/home/ben/code/goprojects/src/scratchpad/jsdefer/main.go.122417206:1480
        throw err;
        ^
null
```

**Comment by grantnelson-wf** [comment]
Here is a slightly simpler version of the same code:
```GO
package main

import "fmt"

func main() {
	err := processData()
	fmt.Println("Got error:", err)
}

func processData() (err error) {
	defer func() {
		if r := recover(); r != nil {
			err = fmt.Errorf("error processing data: %v", r)
		}
	}()

	// openResource()
	defer closeResource()

	return nil
}

func closeResource() {
	panic("some error closing resource")
}
```

Here it is in Go Playground: https://play.golang.org/p/Jobc1JL87Bc
Go playground returns the result:
```
Got error: error processing data: some error closing resource
```

Here it is in GopherJS Playground: https://gopherjs.github.io/playground/#/zPw8fJlyxp
No result is printed but the browser console has this error:
```
angular.js:9937
TypeError: Cannot read property 'message' of null
    at eval (eval at $b (playground.js:77), <anonymous>:2347:20)
    at eval (<anonymous>)
    at $b (playground.js:77)
    at k.v.$externalizeWrapper [as run] (playground.js:4)
    at angular.js:10694
    at angular.js:18882
    at k.$eval (angular.js:12556)
    at k.$apply (angular.js:12654)
    at HTMLInputElement.<anonymous> (angular.js:18881)
    at angular.js:2816
    at Array.forEach (<anonymous>)
    at q (angular.js:320)
    at HTMLInputElement.c (angular.js:2815)
```

The problem that any panic which is emitted from a deferred statement can not be caught with a recover outside of that statement

**Comment by grantnelson-wf** [comment]
As clarification, we've been using this kind of defer recovery in our project for a while and this bug is new to gopherJS as of 1.10

**Comment by grantnelson-wf** [comment]
Just rechecked using the playgrounds (v1.11 ?); This bug is still repeatable

**Comment by grantnelson-wf** [comment]
Still getting the same console error with no output when running the [gopherjs test on the playground](https://gopherjs.github.io/playground/#/zPw8fJlyxp) (go1.16.3). The same test on the [go playground works correctly](https://play.golang.org/p/Jobc1JL87Bc).

**Comment by nevkontakte** [comment]
The playground lags a bit behind the releases, but I was able to reproduce the issue on the latest release. This seems similar to the issue I've fixed in https://github.com/gopherjs/gopherjs/pull/1031, but apparently is different enough to survive that fix :)

`null` output is a smoking gun here. GopherJS throws null to unwind stack after handing a panic. Apparently it doesn't unwind correctly in this case. I'll try to take a look in the next week or two, though most of my attention is currently on supporting Go 1.17.

If you are interested to poke at the issue with the debugger yourself, I will happily answer questions (at least the ones I know answers for 😅) here or in the gopherjs slack channel.

**Comment by grantnelson-wf** [comment]
Awesome, thanks. 😄 👍  I will check out the debugger. I've just been keeping an eye on this one since we still have code in various places which does this, and we haven't been able to rework it all to prevent hitting this case (and we were hoping this bug will eventually be fixed). Fortunately no customers have ever hit those error handlers in all this time so I'm not too worried about it.

**Comment by nevkontakte** [comment]
Good news, I've looked into this bug last night (as I was looking into a similar one I ran into on my experimental branch) and https://github.com/gopherjs/gopherjs/pull/1084 has a fix for it 🙂 Once the pull request is merged, I'm going cut a fresh release, since we have a fair few changes accumulated by now.

**Comment by grantnelson-wf** [comment]
That is great news. Thank you for looking into this.
